pints/_nested/__init__.py

```diff
@@ -9,11 +9,191 @@
 from __future__ import absolute_import, division
 from __future__ import print_function, unicode_literals
 import pints
+import numpy as np
+from scipy.special import logsumexp
 
 
 class NestedSampler(pints.TunableMethod):
     """
     Abstract base class for nested samplers.
+    """
+    def __init__(self, log_prior):
+        # Store function
+        if not isinstance(log_prior, pints.LogPrior):
+            raise ValueError('Given log_prior must extend pints.LogPrior')
+        self._log_prior = log_prior
+        self._running_log_likelihood = -float('Inf')
+        self._proposed = None
+
+        # Initialise active point containers
+        self._n_active_points = 400
+        self._n_parameters = self._log_prior.n_parameters()
+        self._m_active = np.zeros((self._n_active_points,
+                                   self._n_parameters + 1))
+        self._min_index = None
+        self._accept_count = 0
+        self._n_evals = 0
+
+    def needs_sensitivities(self):
+        """
+        Determines whether sampler uses sensitivities of the solution.
+        """
+        return self._needs_sensitivities
+
+    def name(self):
+        """ Name of sampler """
+        raise NotImplementedError
+
+    def ask(self):
+        """
+        Proposes new point at which to evaluate log-likelihood.
+        """
+        raise NotImplementedError
+
+    def tell(self, fx):
+        """
+        Whether to accept point if its likelihood exceeds the current
+        minimum threshold.
+        """
+        # if running in parallel
+        if isinstance(fx, list):
+            a_len = len(fx)
```

Comment:
I can't really judge if this is doing what it's supposed to do! Some more comments here and in the constructor where all properties are defined would be much appreciated

Reply:
Added quite a few comments now